src/shared/services/form/form-value.service.ts

```diff
@@ -218,8 +218,13 @@ export class FormValueService {
     }
 
     let sanitisedObject = {};
+    const documentFieldKeys = ['document_url', 'document_binary_url', 'document_filename'];
     Object.keys(rawObject).forEach(key => {
-      if ('CaseReference' === key) {
+      // If the key is one of documentFieldKeys, it means the field is of Document type. If the value of any of these
+      // properties is null, the entire sanitised object to be returned should be null
+      if (documentFieldKeys.indexOf(key) > -1 && rawObject[key] == null) {
+        sanitisedObject = null;
+      } else if ('CaseReference' === key) {
```

Comment:
is there unit test to cover this form-value.service.ts?

Reply:
@rayliangatos - there isn't an explicit unit test for this (because the `sanitiseObject` function is `private`). However, it should already be covered by the higher-level functional tests in `case-edit-submit.component.spec.ts`. I'll have a look at adding a unit test if possible.